### PR TITLE
fix(aws apiGateway): restApiRootResourceId is required when restApId is configured

### PR DESCRIPTION
**Pull request by sdas13**
Closes: #8515

### CODE REVIEW

File: lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js [hunk 1]
```diff
@@ -42,6 +42,8 @@ module.exports = {
     const events = [];
     const corsPreflight = {};
 
+    this.validateRestApiRootResourceId();
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js:45]: Let's put this logic inline, as this logic is not extensive and there's no re-usability use case, I believe there's no point to delegate it to dedicated method

File: test/unit/lib/plugins/aws/package/compile/events/apiGateway/lib/validate.test.js [hunk 2]
```diff
@@ -1405,6 +1405,25 @@ describe('#validate()', () => {
       },
     });
   });
+
+  it('should throw an error when restApiRootResourceId is not provided with restApiId', () => {
```
Comment (medikoo) [inline test/unit/lib/plugins/aws/package/compile/events/apiGateway/lib/validate.test.js:1409]: We have a policy that all new tests should be constructed via `runServerless` util (see `describe` below, which already hosts some).

Ideally, we should configure a new `runServerless` based test, that confirms that an error with expected code was thrown

File: lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js [hunk 3]
```diff
@@ -470,4 +472,15 @@ module.exports = {
     const splitLambdaName = splitArn[splitArn.length - 1].split('-');
     return splitLambdaName[splitLambdaName.length - 1];
   },
+
+  validateRestApiRootResourceId() {
+    const apiGateway = this.serverless.service.provider.apiGateway;
+
+    if (apiGateway && apiGateway.restApiId && !apiGateway.restApiRootResourceId) {
+      throw new ServerlessError(
+        "Configuration warning at 'provider.apiGateway': should have required property 'restApiRootResourceId'",
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js:481]: We're throwing an _error_, so it's not a warning (which is permissive by nature).

Let's reword it to:

```
Missing required "provider.apiGateway.restApiRootResourceId" property (needed if "provider.apiGateway.restApiId" is provided)
```

File: lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js [hunk 4]
```diff
@@ -470,4 +472,15 @@ module.exports = {
...
+        'RESTAPIROOTRESOURCEID_PROPERTY_MISSING'
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js:482]: Let's change it to: `API_GATEWAY_MISSING_REST_API_ROOT_RESOURCE_ID`

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @sdas13 ! It looks very good, I've proposed just a few improvements. Let me know

File: lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js (hunk 1 above)
Comment (sdas13) [reply lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js]: yes. makes sense

**Comment by sdas13** [comment]
@medikoo will create a new PR. This one got messed up due to wrong merging in my local
